### PR TITLE
Rework WrapSolidify effect

### PR DESCRIPTION
**Pull request by lassoan**
I've worked a lot on understanding and improving/adding features to this effect. Algorithm of how cavities are curved out has been reworked, as it was not very robust for complex shapes, but otherwise all the core concepts and features have been preserved.

Highlights:

- Create GUI using Qt Designer
- Add automatic cavity extraction option
- Add option to specify starting surface (to allow users to solidify any part of the segment)
- Added new algorithm for carving out cavities very robustly from arbitrarily complex shapes
- Removed a few hardcoded values (they are now computed from inputs)

It would be great if you could test it on your data sets and let me know how it works. If it looks all good then please merge this pull request.

### CODE REVIEW

**Comment by lassoan** [comment]
@sebastianandress I've worked a _lot_ on this and would really like to make the new features and improvements available to Slicer users. When do you think you will have time to check this out?

If you prefer to keep your segment editor effect as is and have a separate effect for general use then we can do that, too, but the two modules would be somewhat redundant and there would be overall maintenance workload.

Anyway, let me know how you want to proceed.

Thanks again for your very useful and important contribution!

**Comment by sebastianandress** [comment]
Dear @lassoan,
I‘m amazed by the effort you put into improving the effect. I‘ll test everything on the weekend and merge it! Thanks a lot!

**Comment by lassoan** [comment]
Sounds good, thank you!

File: SegmentEditorWrapSolidify/SegmentEditorWrapSolidifyLib/SegmentEditorEffect.py [hunk 1]
```diff
@@ -234,854 +127,816 @@ def processViewNodeEvents(self, callerViewNode, eventId, viewWidget):
...
-
-    shrinkModelPD = vtk.vtkPolyData()
+    cleanPolyData.SetInputData(surfacePd)
+
+    # create normals
+    normals = vtk.vtkPolyDataNormals()
+    normals.SetComputeCellNormals(1)
+    normals.SetInputConnection(cleanPolyData.GetOutputPort())
+    normals.SplittingOff()
+    if shellOffsetDirection == SHELL_OFFSET_INSIDE:
```
Comment (sebastianandress) [inline SegmentEditorWrapSolidify/SegmentEditorWrapSolidifyLib/SegmentEditorEffect.py:770]: in my tests that's wrong way round, probably this has to be SHELL_OFFSET_OUTSIDE

**Review by sebastianandress — CHANGES_REQUESTED** [review]
Dear @lassoan, thank you for these improvements. I really like that the model gets shrunken to the dilated model, it's a nice solution for my prototype approach. Also the option to start from a dedicated segment is very helpful for some applications.

The only thing that is flipped around in my eyes is the offset direction. Besides that I tested the changes for about 10 models and it worked for all of them. Thanks a lot!

**Comment by lassoan** [comment]
Great, thanks for testing! It would be great if you could merge it and we can update how the offset direction is specified in a follow-up commit.

**Pull request merged**